batch.R

```diff
@@ -108,10 +98,18 @@ for(i in 1:nrow(fileDF)) {
   qColName <- names(siteConstit)[2]
   dateColName <- names(siteConstit)[1]
   
+  # remove duplicate observations
+  constitDupes <- table(siteConstit$date) %>% .[.>1] %>% names()
+  nonFirstDupes <- unlist(lapply(constitDupes, function(cD) {
+    dupes <- which(as.character(siteConstit$date) == cD)
+    dupes[-1]
```

Comment:
The line after this deletes all elements in `nonFirstDupes`. So I omit the first duplicate in each set in `nonFirstDupes` so that we can keep that first copy of each duplicated observation in `siteConstit`. Then we're only deleting information that's redundant with what we're keeping in the dataset.